pkg/buildpack/module_manager.go

```diff
@@ -0,0 +1,144 @@
+package buildpack
+
+import (
+	"github.com/buildpacks/pack/pkg/dist"
+)
+
+const (
+	FlattenMaxDepth = -1
+	FlattenNone     = 0
+)
+
+type ModuleManager struct {
+	modules        []BuildModule
+	flattenModules [][]BuildModule
+	flatten        bool
+	maxDepth       int
+}
+
+func NewModuleManager(flatten bool, maxDepth int) *ModuleManager {
+	return &ModuleManager{
+		flatten:        flatten,
+		maxDepth:       maxDepth,
+		modules:        []BuildModule{},
+		flattenModules: [][]BuildModule{},
+	}
+}
+
+// AllModules returns all modules handle by the manager
+func (f *ModuleManager) AllModules() []BuildModule {
+	all := f.modules
+	for _, modules := range f.flattenModules {
+		all = append(all, modules...)
+	}
+	return all
+}
+
+// NonFlattenModules returns all none flatten modules handle by the manager
+func (f *ModuleManager) NonFlattenModules() []BuildModule {
```

Comment:
```suggestion
// ExplodedModules returns all modules that will be added to the output artifact as a single layer containing a single module.
func (f *Collection) ExplodedModules() []BuildModule {
```
Is that accurate? I think a slightly more descriptive comment would help others to understand it better.